examples/pipelines/controlnet-interior-design/components/retrieve_images/fondant_component.yaml

```diff
@@ -0,0 +1,27 @@
+name: Retrieve images
+description: A component that retrieves image URLs from LAION-5B based on a set of seed prompts
+image: generate_prompts::latest
+
+input_subsets:
+  prompts:
+    fields:
+      text:
+        type: utf8
+
+output_subsets:
+  images:
+    fields:
+      url:
+        type: utf8
+  additionalSubsets: false
+
+args:
+  num_images:
+    description: Number of images to retrieve for each prompt
+    type: int
+  aesthetic_score:
+    description: Ranking score for aesthetic, higher is prettier
+    type: int
+  aesthetic_weight:
+    description: Weight of the aesthetic score, between 0 and 1
```

Comment:
what's the difference between the weight and the score?

Reply:
CLIP-retrieval uses [this model](https://github.com/LAION-AI/aesthetic-predictor) to do the following operation:
```
aesthetic_embedding = clip_resource.aesthetic_embeddings[aesthetic_score]
query = query + aesthetic_embedding * aesthetic_weight
query = query / np.linalg.norm(query)
```